### ISSUE TITLE
add validation

### ISSUE THREAD

**Issue by viktoriasee** [peiTeX/qrbill#7]
### Requirements
There are lots of requirements for data in a QR payment slip. Many of them are not obvious. I.e. you should write amounts with two decimal places which requires trailing zeros: 45.50 instead of 45.5

### Current behaviour
Currently even the demo example fails validation in https://www.swiss-qr-invoice.org/validator/ Data seems to be unchecked. If someone enters the IBAN with spaces this leads to an invalid QR code.

### Proposed solution
Validate the entries whereever possible.

**Comment by TeXhackse** [comment]
The example data fails due to #3. We are aware of this, and it will be resolved with the update for the support of versions 2.2.
This is actually not a bug in the qrbill package but in the standard version 2.1  We reported it and they finally managed to release the update. 

Anyway I will think about this. Generally qrbill wanted to be a wrapper for the data and simplify the output. It's possible to add validation, though the current usecases use qrbill together with automatic generated invoices which do this validation independently in advance. Are you planning to fill these manually? I just would like to understand your usecase. Will think about it. There are some rules which are too specifc to add them to then to the package in general (like only allow CH & LI Ibans), but maybe it's an option to add some rules to allow the user to control validation.

**Comment by viktoriasee** [comment]
I think the example fails because the VAT number checksum is invalid. That has nothing to do with #3 .

My use case is to manually write letters with a payment slip attached. This is needed if you need to be reimbursed i.e. you bought a flight but it's being cancelled due to COVID. Or you bought a house but have to pay for the oil in the tank separately. Or you sell a car.

Is is maybe possible to directly use the [Validator from Github](https://github.com/swico/www.swiss-qr-invoice.org/)?

**Comment by TeXhackse** [comment]
Yup, you are right. But I sadly don't have a swiss VAT number to enter there. So we will keep it that way. 

I see. Well we wanted to develop qrbill as an offline solution especially since there is a bunch of online tools to create qrbills but one cannot say anything about their data protection. The website you linked is based on Version 1.2 of the QRbill. So it's actually not the last version. I didn't check the differences, But I'd prefer something which uses the last version of the guidelines, because the issue we had with 0210 already was caused by a but in most implementations of the qrbill guidelines. The repo you linked does not contain any validator, just the website wrapping the validator. It's a JS script, one could ofc run on the data in advance, but I currently don't have enough free time to add this.

Feel free to provide a lua or TeX base Script doing the validation. I can also try to help you.

Anyway I think qrbill might contain an IBAN validation, though it's country specific and we'd then have to implement all of them. 
But usually you will use your own data when writing these letters, wouldn't you? So the validation would only be required once, because you will reuse it. Or did I get this wrong?

**Comment by ondohotola** [comment]
https://www.subtechsports.com/wp-content/uploads/2020/12/gulsps.jpg

**Comment by ondohotola** [comment]
[Rosetta](https://rosettacode.org/wiki/IBAN) has a LUA IBAN validation
